### PR TITLE
prov/efa: various small changes

### PR DESCRIPTION
**Pull request by wzamazon**
Mainly to clean up `efa.h` and `rxr.h`

### CODE REVIEW

File: prov/efa/src/rdm/rxr_env.c [hunk 1]
```diff
@@ -83,6 +82,12 @@ void rxr_env_param_get(void)
 		abort();
 	};
 
+	if (getenv("FI_EFA_MTU_SIZE")) {
+		fprintf(stderr,
+			"FI_EFA_MTU_SIZE env variable detected! The use of this variable has been deprecated and as such execution cannot proceed.\n");
+		abort();
```
Comment (sunkuamzn) [inline prov/efa/src/rdm/rxr_env.c:88]: Do we have to abort if the env variable is set?

File: prov/efa/src/rdm/rxr_ep.c [hunk 2]
```diff
@@ -794,15 +794,6 @@ static int rxr_ep_ctrl(struct fid *fid, int command, void *arg)
 
 		rxr_ep_set_extra_info(ep);
 
-		ep->core_addrlen = RXR_MAX_NAME_LENGTH;
```
Comment (sunkuamzn) [inline prov/efa/src/rdm/rxr_ep.c:797]: I think the commit to remove `RXR_MAX_NAME_LENGTH` can be squashed into this commit

File: prov/efa/src/rdm/rxr_env.c (hunk 1 above)
Comment (wzamazon) [reply prov/efa/src/rdm/rxr_env.c:88]: Yes. that is what we did for other deprecated env vars.

File: prov/efa/src/rdm/rxr_ep.c (hunk 2 above)
Comment (wzamazon) [reply prov/efa/src/rdm/rxr_ep.c:797]: Rebase is a little tricky here because commits modified same file `rxr.h`. Do you have strong preference?

Comment (sunkuamzn) [reply prov/efa/src/rdm/rxr_ep.c:797]: No strong preference

**Review by sunkuamzn — APPROVED** [review]

File: prov/efa/src/rdm/rxr_env.c (hunk 1 above)
Comment (shijin-aws) [reply prov/efa/src/rdm/rxr_env.c:88]: I was wondering why we could make it customized by envs.

**Review by shijin-aws — APPROVED** [review]

File: prov/efa/src/rdm/rxr_env.c [hunk 3]
```diff
@@ -83,14 +82,21 @@ void rxr_env_param_get(void)
...
-			"FI_EFA_TX_MIN_CREDITS was set to %d, which is <= 0."
-			"This value will cause EFA communication to deadlock."
-			"To avoid that, the variable was reset to %d\n",
-			rxr_env.tx_min_credits, RXR_DEF_MIN_TX_CREDITS);
-		rxr_env.tx_min_credits = RXR_DEF_MIN_TX_CREDITS;
+		fprintf(stderr,
+			"FI_EFA_TX_MIN_CREDITS was set to %d, which is <= 0.\n"
+			"This value will cause EFA communication to deadlock.\n"
+			"Please unset the environment variable or set it to a positive number\n"
+			"You application will now abort.",
```
Comment (darrylabbate) [inline prov/efa/src/rdm/rxr_env.c:97]:
```suggestion
			"Please unset the environment variable or set it to a positive number.\n"
			"Your application will now abort.",
```

No strong preference on the punctuation, but it should be consistent.

File: prov/efa/src/rdm/rxr_env.h [hunk 4]
```diff
@@ -34,6 +34,12 @@
 #ifndef _RXR_ENV_H
 #define _RXR_ENV_H
 
+/**
+ * Setting ibv_qp_attr.rnr_retry to this number when modifying qp
+ * to cause firmware to retry indefininetly.
```
Comment (darrylabbate) [inline prov/efa/src/rdm/rxr_env.h:39]:
```suggestion
 * to cause firmware to retry indefinitely.
```

(I know this was copy/pasted as-is)

File: prov/efa/src/rdm/rxr_env.c (hunk 1 above)
Comment (wzamazon) [reply prov/efa/src/rdm/rxr_env.c:88]: > I was wondering why we could make it customized by envs.

It was introduced back when RxR provider was an utility provider that operates on another RDM provider.

File: prov/efa/src/rdm/rxr_ep.c [hunk 5]
```diff
@@ -2193,14 +2131,7 @@ int rxr_endpoint(struct fid_domain *domain, struct fi_info *info,
...
 
-	ret = rxr_ep_alloc_app_device_info(&rdm_info,
-					   efa_domain->device->rdm_info,
-					   efa_domain->util_domain.fabric->fabric_fid.api_version
-					   );
-	if (ret)
-		goto err_close_ofi_ep;
-
-	ret = efa_base_ep_construct(&rxr_ep->base_ep, rdm_info);
+	ret = efa_base_ep_construct(&rxr_ep->base_ep, info);
```
[the remark this answers is not in the document]
Comment (wzamazon) [reply prov/efa/src/rdm/rxr_ep.c]: Yes.

File: prov/efa/src/rdm/rxr_env.h (hunk 4 above)
Comment (wzamazon) [reply prov/efa/src/rdm/rxr_env.h]: Fixed in new revision.

**Review by darrylabbate — APPROVED** [review]

**Pull request merged**